analysis/topology/sequences.lean

```diff
@@ -0,0 +1,282 @@
+iff.intro
+  (assume _, cont_to_seq_cont ‹continuous f›) 
+  (assume : sequentially_continuous f, show continuous f, from
+    suffices h : ∀ {A : set Y}, is_closed A → is_seq_closed (f ⁻¹' A), from 
+      continuous_iff_is_closed.mpr (assume A _, 
+        is_seq_closed_iff_is_closed.mp $ h ‹is_closed A›),
+    assume A (_ : is_closed A),
+      is_seq_closed_of_def $
+        assume (x : ℕ → X),
+        assume :∀ n, (x n) ∈ (f ⁻¹' A),
+        have ∀ n, f (x n) ∈ A, by assumption,
+        assume p (_: converges_to x p),
+        have converges_to (f ∘ x) (f p), from ‹sequentially_continuous f› x ‹converges_to x p›, 
+        have f p ∈ A, from is_mem_of_is_closed_of_conv_to ‹is_closed A› ‹∀ n, f (x n) ∈ A›
+          ‹converges_to (f∘x) (f p)›,
+        this)
+  
+
+end topological_space
+
+/- Statements about sequences in metric spaces -/
+section metric_space
+variable [metric_space X]
+variables {ε : ℝ}
+
+/- The usual notion of convergence of sequences in metric spaces. -/
+def metrically_converges_to (x : ℕ → X) (limit : X) : Prop :=
+  ∀ ε > 0, ∃ n0 : ℕ, ∀ n ≥ n0, dist (x n) limit < ε
+
+/- A sequence converges metrically if and only if it converges topologically. -/
+lemma metrically_converges_to_iff_converges_to {x : ℕ → X} {limit : X} :
+  metrically_converges_to x limit ↔ converges_to x limit :=
+iff.intro
+  (assume metrConvTo,
+    show ∀ U : set X, limit ∈ U → is_open U → ∃ n0 : ℕ, ∀ n ≥ n0, (x n) ∈ U, from
+    assume U limitInU isOpenU,
+      have ∃ ε > 0, ball limit ε ⊆ U ,from is_open_metric.mp isOpenU limit limitInU,
+      let ⟨ε, _, _⟩ := this in
+      have ∃ n0, ∀ n ≥ n0, dist (x n) limit < ε, from ‹metrically_converges_to x limit› ε ‹ε > 0›, 
+      let ⟨n0, _⟩ := this in
+      show ∃ n0, ∀ n ≥ n0, (x n) ∈ U, from  
+        exists.intro n0 (assume n _, 
+          have (x n) ∈ ball limit ε, from ‹∀ n ≥ n0, dist (x n) limit < ε› _ ‹n ≥ n0›,
+          show (x n) ∈ U, from set.mem_of_subset_of_mem ‹ball limit ε ⊆ U› ‹(x n) ∈ ball limit ε›))
+  (assume convTo,
+    show ∀ ε > 0, ∃ n0, ∀ n ≥ n0, dist (x n) limit < ε, from
+    assume ε _,
+    have ∃ n0 : ℕ, ∀ n ≥ n0, (x n) ∈ (ball limit ε), from 
+      ‹converges_to x limit› (ball limit ε) (mem_ball_self ‹ε > 0›) is_open_ball,
+    let ⟨n0, _⟩ := this in
+    exists.intro n0 
+      (show ∀ n ≥ n0, dist (x n) limit < ε, from
+        assume n _,
+          ‹∀ n ≥ n0, (x n) ∈ ball limit ε› n ‹n ≥ n0›))
+
+/- A sequence converges metrically iff the associated statement for filters holds true. -/
+lemma metrically_converges_to_iff_tendsto {x : ℕ → X} {limit : X} :
+  metrically_converges_to x limit ↔ tendsto x at_top (nhds limit) :=
+calc metrically_converges_to x limit ↔ converges_to x limit :
+                                                      metrically_converges_to_iff_converges_to
+                                 ... ↔ tendsto x at_top (nhds limit) : converges_to_iff_tendsto
+
+private lemma one_div_succ_pos (n : ℕ) : 1 / ((n : ℝ) + 1) > 0 :=
```

Comment:
I think I did this originally, but this should probably be moved somewhere more general.

Reply:
`algebra/ordered_field.lean`?

Reply:
That seems wrong as it would imply importing the reals in ordered field.

Reply:
This should work more generally for some ordered fields